File: 2017/05.js

const fs = require('fs');
const { exec } = require("child_process");
const { ALPHA_L, ALPHA_U, D4, D6, D8, MOD } = require("../lib");
const { joseph, findCycle } = require("../lib");
const { cache } = require("../lib");
const { bin, float, hex, int, num, oct } = require("../lib");
const { range, drange, trange, iter, diter, titer } = require("../lib");
const { copy,	entries, in_, inBB, keys, sort, values } = require("../lib");
const { Counter, DD, empty, FastQueue, PriorityQueue, Queue, set, Stack } = require("../lib");
const { areaInt, circumference, manDist, shoelace } = require("../lib");
const { ord, chr, count, debug, disp, disp3, crt, gcd, lcm, modPow, modPowBig, modInv, mod, prod, prodBig, randint, sum, sumBig, transpose } = require("../lib");
const { digits, ints,	floats,	singles,	words,	lines,	table,	groups,	getGroups,	groupsWith,	parse,	parseLine, scanf } = require ('../lib');
const { min, max, random, abs, ceil, floor, log, log10, log2, round, sign, sin, cos, tan, asin, acos, atan, atan2, sqrt, PI } = Math;
const { isSuperset, or, and, xor, sub } = set;
const { getExecStr } = require("../lib/post");
const { combinations, combinations_with_replacement, next_permutation, product } = require("../lib");

function part1(data) {

	let res = 0;
	data = ints(data);
	let curr = 0;
	while (curr >= 0 && curr < data.length) {
		const step = data[curr];
		data[curr]++;
		curr += step;
		++res;
	}

	debug(res);
	if (res) exec(`echo ${res} | xclip -sel clip -rmlastnl`);
	console.log("END OF PART1");
	return;
}

function part2(data) {

	let res = 0;
	data = ints(data);
	let curr = 0;
	while (curr >= 0 && curr < data.length) {
		const step = data[curr];
		if (step >= 3) data[curr]--;
		else data[curr]++;
		curr += step;
		++res;
	}
	
	debug(res);
	if (res) exec(`echo ${res} | xclip -sel clip -rmlastnl`);
	console.log("END OF PART2");
	return;
}

function main() {
	let data = fs.readFileSync("05_input").toString("utf-8");

	part1(data);
	part2(data);
	process.exit(0);
}

main();

File: 2017/05.py

import math, copy, re


def part_1(data):
	steps = 0
	i = 0
	L = len(data)
	while i >= 0 and i < L:
		curr = data[i]
		data[i] += 1
		i += curr
		steps += 1
	print(steps)
	return

def part_2(data):
	steps = 0
	i = 0
	L = len(data)
	while i >= 0 and i < L:
		curr = data[i]
		if curr >= 3: data[i] -= 1
		else: data[i] += 1
		i += curr
		steps += 1
	print(steps)
	return 


if __name__ == '__main__':
	with open('5_input') as f:
		data = f.read()
		# data = data.split('\n')
		data = list(map(int, data.split()))


	part_1(data)
	part_2(data)
	